singer_sdk/target_base.py

```diff
@@ -400,6 +402,15 @@ def _process_activate_version_message(self, message_dict: dict) -> None:
         sink = self.get_sink(stream_name)
         sink.activate_version(message_dict["version"])
 
+    def _process_batch_message(self, message_dict: dict) -> None:
+        """Handle the optional BATCH message extension.
+
+        Args:
+            message_dict: TODO
+        """
+        encoding = BaseBatchFileEncoding.from_dict(message_dict["encoding"])
+        self.logger.info("Processing record batch encoded as %s", encoding)
```

Comment:
As a default implementation, will we also be passing records to `Target._process_record_message(message_dict)`?

It appears as of now that this is still a noop; just wanted to confirm.

Reply:
Looking closer, actually this probably needs some more refactoring, since `message_dict` in the above is expecting a singer RECORD message - and the logic in that loop is pretty in-depth. 

Perhaps a new and common `_process_record(message_dict, stream_name)` could be called by both `_process_record_message()` and `_process_batch_message()`?

Reply:
> Perhaps a new and common `_process_record(record_dict, stream_name)` (or similar) could be called by both `_process_record_message()` and `_process_batch_message()`?

@aaronsteers I have a `Sink.process_batch_file` in my git stash that is called here, but I wanted to be kept honest on the tap implementation first 🙂